### PR TITLE
Refactor(eos_designs): Move eos_designs Python modules to PyAVD

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Move eos_designs Python modules to PyAVD

## Proposed changes

- Move all eos_designs python modules from Ansible collection to PyAVD
  - Adjust imports accordingly
- Split pylint into an ansible-specific pylintrc and a more standard pylintrc for everything else
  - Update pre-commit
  - Fix various pylint issues in PyAVD code
- Fixed bug for MSDP peerings where the wrong address was being used (clearly visible in our test artifacts!)

## How to test

No changes to molecule except the MSDP bugfix.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -1,131 +1,110 @@
...
+
 
 [MESSAGES CONTROL]
 disable=
-    abstract-method,
-    access-member-before-definition,
-    arguments-differ,
-    assignment-from-no-return,
-    assignment-from-none,
+    ; abstract-method,
```
Comment (gmuloc) [inline pylintrc]: let's remove the commented ones

File: python-avd/pyavd/_eos_designs/structured_config/base/__init__.py [hunk 2]
```diff
@@ -5,13 +5,12 @@
...
 from functools import cached_property
 
-from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
-from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
-from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
-from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdMissingVariableError
-from ansible_collections.arista.avd.plugins.plugin_utils.strip_empties import strip_null_from_data
-from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
-
+from ....j2filters.convert_dicts import convert_dicts
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/structured_config/base/__init__.py:8]: let's see if we can change the `....`

File: python-avd/pyavd/_eos_designs/structured_config/overlay/utils.py [hunk 3]
```diff
@@ -273,7 +272,7 @@ def _stun_server_profile_name(self, wan_route_server_name: str, path_group_name:
         return f"{path_group_name}-{wan_route_server_name}-{sanitized_interface_name}"
 
     @cached_property
-    def _stun_server_profiles(self) -> list:
+    def _stun_server_profiles(self) -> dict:
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/structured_config/overlay/utils.py]:
```suggestion
    def _stun_server_profiles(self: AvdStructuredConfigOverlay) -> dict:
```

File: python-avd/pyavd/_eos_designs/structured_config/base/__init__.py (hunk 2 above)
Comment (gmuloc) [reply python-avd/pyavd/_eos_designs/structured_config/base/__init__.py:8]: let's do this in a later PR

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]
LGTM

**Pull request merged**